packages/relay-runtime/util/getRefetchMetadata.js

```diff
@@ -58,14 +62,17 @@ function getRefetchMetadata(
       'this is likely a bug in Relay.',
     componentDisplayName,
   );
-  const identifierField = refetchMetadata.identifierField;
-  invariant(
-    identifierField == null || typeof identifierField === 'string',
-    'Relay: getRefetchMetadata(): Expected `identifierField` to be a string.',
-  );
+  const identifierInfo = refetchMetadata.identifierInfo;
+  if (identifierInfo) {
+    invariant(
+      identifierInfo.identifierField == null ||
+        typeof identifierInfo.identifierField === 'string',
+      'Relay: getRefetchMetadata(): Expected `identifierField` to be a string.',
+    );
```

Comment:
Should we add a similar invariant for `identifierQueryVariableName`?

Reply:
Sure, it doesn't hurt and it's consistent with the rest of the function. Added!